crates/dojo/core/src/world/storage.cairo

```diff
@@ -32,10 +32,15 @@ pub impl WorldStorageInternalImpl of WorldStorageTrait {
         WorldStorage { dispatcher: world, namespace_hash }
     }
 
+    fn new_from_hash(world: IWorldDispatcher, namespace_hash: felt252) -> WorldStorage {
+        WorldStorage { dispatcher: world, namespace_hash }
+    }
+
```

Comment:
If we add this to save the Poseidon hashes on the serialized byte array, we should also add one for the `dns`. Cairo doesn't support function overloading.

Since computing the dojo selector requires three steps:
1. Hashing the namespace (saved by the `new_from_hash`).
2. Hashing the name.
3. Hashing them together.
Having something like:
```rust
let (ca, ch) = world.dns_from_hash/selector(selector_from_tag!("eternum-map_system")).unwrap();
```
Will be the most efficient way to query a resource, with no Poseidon hash at all.